src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -2456,5 +2457,49 @@ public class C
                 Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(5, 49)
                 );
         }
+
+        [Fact]
+        [WorkItem(58073, "https://github.com/dotnet/roslyn/issues/58073")]
+        public void DiagnosticAdditionalLocations_SquiggleConstructor()
+        {
+            var source =
+@"
+public class C
+{
+    public C() { }
+
+    public string S { get; }
+}";
+            var comp = CreateCompilation(source, options: WithNullableEnable());
+            comp.VerifyDiagnostics(
+                // (4,12): warning CS8618: Non-nullable property 'S' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+                //     public C() { }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("property", "S").WithLocation(4, 12));
```

Comment:
It's slightly unfortunate that this generated baseline doesn't include the additional locations.